app/src/main/kotlin/io/intrepid/fablekotlin/screens/createfable/CreateFableActivity.kt

```diff
@@ -70,6 +105,117 @@ class CreateFableActivity : BaseMvpActivity<CreateFableContract.Presenter>(), Cr
+    }
+
+    override fun onOptionsItemSelected(item: MenuItem): Boolean {
+        val title = enteredFableTitle.text.toString()
+        val itemClicked = item.itemId
+        when (itemClicked) {
+            R.id.createFable -> {
+                presenter.onContinueClicked(title)
+                return true
+            }
+        }
+        return super.onOptionsItemSelected(item)
+    }
+
+    override fun onSupportNavigateUp(): Boolean {
+        startActivity(Intent(this, HomescreenActivity::class.java))
+        return false
+    }
+
+    // Grabs stored invitees from the other screen, fills the circles with their profile pictures
+    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
+        super.onActivityResult(requestCode, resultCode, data)
+        if (requestCode == PASS_FABLE_INFO) {
+            if (resultCode == Activity.RESULT_OK) {
+                if (data != null) {
+                    selectedFriends = data.getSerializableExtra(SELECTED_FRIENDS) as java.util.ArrayList<GetUserFriendsResponse.Friend>
+                }
+                presenter.setSelectedFriends(selectedFriends)
+                for (i in 2..5) {
+                    setCircleImage(i, null)
+                }
+                for (i in selectedFriends.indices) {
+                    setCircleImage(i + 1, selectedFriends[i].image)
+                }
+            }
+            if (resultCode == Activity.RESULT_CANCELED) {
+                showErrorMessage()
+            }
+        }
+    }
+
+    override fun showTitleShortMessage() {
+        val snackbar = Snackbar.make(findViewById(R.id.newFableLayout),
+                R.string.no_title_warning,
+                Snackbar.LENGTH_SHORT)
+        setSnackTextColor(snackbar.view)
+        snackbar.show()
+    }
+
+    override fun showTitleLongMessage() {
+        val snackbar = Snackbar.make(findViewById(R.id.newFableLayout),
+                R.string.long_title_warning,
+                Snackbar.LENGTH_LONG)
+        setSnackTextColor(snackbar.view)
+        snackbar.show()
+    }
+
+    override fun showNotEnoughFriendsMessage() {
+        val snackbar = Snackbar.make(findViewById(R.id.newFableLayout),
+                R.string.invite_more_friends,
+                Snackbar.LENGTH_LONG)
+        setSnackTextColor(snackbar.view)
+        snackbar.show()    }
+
+    override fun goToFirstSentenceScreen() {
+        val intent = Intent(this, FirstSentenceActivity::class.java)
+        intent.putExtra(FABLE_TITLE, enteredFableTitle.text.toString())
+        intent.putExtra(COLOR_THEME, presenter.getColorTheme())
+        intent.putExtra(ICON, (pager.currentItem + 1).toString())
+        intent.putExtra(SELECTED_FRIENDS, presenter.getSelectedFriends() as Serializable)
+        startActivity(intent)
+    }
+
+    //Set a circle to a profile image
+    override fun setCircleImage(idx: Int, imageSrc: String?) {
+        if (idx < 0 || idx >= circleButtons.size) {
```

Comment:
This might be useful as a static helper/utility method, I think its fine to leave it this way for now.